### PR TITLE
History bug fix

### PR DESCRIPTION
**Pull request by psbritt**
Fixed an issue where only a limited amount (~2^32) histories were able to be performed with a variable type change.

### CODE REVIEW

**Comment by psbritt** [comment]
@gonuke I think should only be 1 commit on this. Since I just branched this from my main branch, I'm guessing I accidentally pushed to my main in October of 2020 and tried to revert it in a way I shouldn't have. I'm not sure how to fix that, but the only files changed appear to be the ones that needed to be changed.

File: packages/monte_carlo/event/core/test/tstParticleHistorySimulationCompletionCriterion.cpp [hunk 1]
```diff
@@ -41,21 +41,25 @@ FRENSIE_UNIT_TEST( HistoryCountParticleHistorySimulationCompletionCriterion,
 FRENSIE_UNIT_TEST( HistoryCountParticleHistorySimulationCompletionCriterion,
                    collect_data )
 {
+
+  uint64_t history_wall = 5000000000;
+
   std::shared_ptr<MonteCarlo::ParticleHistorySimulationCompletionCriterion>
-    criterion = MonteCarlo::ParticleHistorySimulationCompletionCriterion::createHistoryCountCriterion( 10 );
+    criterion = MonteCarlo::ParticleHistorySimulationCompletionCriterion::createHistoryCountCriterion( history_wall );
+
```
Comment (lewisgross1296) [inline packages/monte_carlo/event/core/test/tstParticleHistorySimulationCompletionCriterion.cpp:49]: maybe add a comment here to why you picked the history wall to be `5e9` for future people looking at this

**Comment by lewisgross1296** [comment]
I am assuming all tests passed (so this may be redundant given your test modified in this PR), but I can modify my copy of `FRENSIE/packages/monte_carlo/event/core/src/MonteCarlo_ParticleHistorySimulationCompletionCriterion.cpp` and run a `5e9` particle simulation to see if it actually finishes. I will request 5 rendezvous points for this one since we'd expect it to fail between `4e9` and `5e9` particles if the bug is still present. I can report back on this after it hopefully finishes successfully.

**Review by gonuke — APPROVED** [review]
This LGTM - sure would be nice if we had CI running... maybe with docker images and github actions?

**Comment by gonuke** [comment]
Should I be waiting on any final testing, in the absence of CI?

**Comment by gonuke** [comment]
Thanks @psbritt

**Pull request merged**